ecs-agent/config/interface.go

```diff
@@ -0,0 +1,62 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package config
+
+// AgentConfigAccessor provides access to an agent config via a set of designated methods (listed below).
+type AgentConfigAccessor interface {
+	// APIEndpoint returns the endpoint to make calls against.
+	// (e.g., "ecs.us-east-1.amazonaws.com")
+	APIEndpoint() string
+	// AWSRegion returns the region to run in.
+	// (e.g., "us-east-1")
+	AWSRegion() string
+	// Cluster returns the name or full ARN of the cluster that the Agent
+	// should register the container instance into.
+	Cluster() string
+	// DefaultClusterName returns the name of the cluster that the container
+	// instance should be registered into by default (i.e., if no cluster is
+	// specified).
+	// (e.g., "default")
+	DefaultClusterName() string
+	// External returns whether the Agent is running on external compute
+	// capacity (i.e., outside of AWS).
+	External() bool
+	// InstanceAttributes returns a map of key/value pairs representing
+	// attributes to be associated with the instance within the ECS service.
+	// They are used to influence behavior such as launch placement.
+	InstanceAttributes() map[string]string
+	// NoInstanceIdentityDocument returns whether the Agent should register
+	// the instance with instance identity document.
+	NoInstanceIdentityDocument() bool
```

Comment:
nit: What does the prefix 'No' signify here? - `NoInstanceIdentityDocument`. Can the name be changed to something like `InstanceIdentityDocumentNeeded` which might make it more clear what the method returns? Otherwise, maybe we can update the comment above with what does it will mean when this returns `true`, i.e. if the method returns true, should Agent register with Instance Identity Document.

Reply:
The 'No' signifies the instance identity document is not available or needed (e.g., when Agent is running on external compute capacity). This mirrors the [agent config field NoIID defined here](https://github.com/aws/amazon-ecs-agent/blob/master/agent/config/types.go#L285-L289). I believe we want the 'No' in the name because the zero/default value of values of type `bool` is `false`, which matches our expectation (i.e., by default, unless otherwise specified, we would expect instance identity document to be available or needed).

I will update the comment on L39/L40 to clarify what it means when the `NoInstanceIdentityDocument` method returns true. Thanks for your suggestion!